modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/raft/ZonePartitionRaftListener.java

```diff
@@ -268,13 +272,24 @@ private IgniteBiTuple<Serializable, Boolean> processCrossTableProcessorsCommand(
     ) {
         IgniteBiTuple<Serializable, Boolean> result = new IgniteBiTuple<>(null, false);
 
-        tableProcessors.values().forEach(processor -> {
-            IgniteBiTuple<Serializable, Boolean> r = processor.processCommand(command, commandIndex, commandTerm, safeTimestamp);
-            // Need to adjust the safe time if any of the table processors successfully handled the command.
-            if (Boolean.TRUE.equals(r.get2())) {
-                result.set2(Boolean.TRUE);
+        if (tableProcessors.isEmpty()) {
+            synchronized (tableProcessorsStateLock) {
+                if (lastAppliedIndex < commandIndex) {
+                    lastAppliedIndex = commandIndex;
+                    lastAppliedTerm = commandTerm;
+
+                    result.set2(Boolean.TRUE);
```

Comment:
Can be replaced with just `return new IgniteBiTuple<>(null, lastAppliedIndex < commandIndex)`;

Reply:
Done